### DISCUSSION TITLE
Why does accuracy need argmax and not loss function when there are multiple class output probabilities?

### DISCUSSION THREAD

**Discussion by sohumgautam17** [mrdbourke/pytorch-deep-learning#907] [General]
Why does accuracy get argmax and loss not? This is in the computer vision section of the course @ 16:48 but its also throughout that section. Heres the code snippet:

 y_eval = model(X)
      loss+=loss_fn(y_eval, y)
      acc = accuracy_fn(y, y_eval.argmax(dim=1))

This is the rest of the training loop
"""
torch.manual_seed(42)
def evalmodel(model:torch.nn.Module, dataloader:torch.utils.data, loss_fn:torch.nn.Module, accuracy_fn):
  loss, acc = 0, 0
  torch.eval()

  with torch.inference_mode():
    for X, y in dataloader:

      y_eval = model(X)
      loss+=loss_fn(y_eval, y)
      acc = accuracy_fn(y, y_eval.argmax(dim=1))

    loss /= len(dataloader) # get the average
    acc /= len(dataloader)

    return{"model_name": model.__class__.__name__,
           "model_Loss": loss.item(), 
           "model_acc" : acc}

"""

**Comment by LuluW8071** [comment]
In background, 
- A loss function measures the difference between the actual values and the predictions made by model.
Example:
  ```python
  5-3=2  # where '5' is actual and `3` is predicted value and `2` is distance between them (called loss)
  ```

- Accuracy is calculated by comparing the predicted labels against the actual labels. Specifically for each prediction, it checks if index of the maxm. value in predicted probabilities corresponds to the index of the actual label.
Example:
  Suppose `[0.1, 0.7, 0.2]` for three classes generated by training a model. 
  ```python
  0.1 # index0
  0.7 # index1
  0.2 # index2
  ```
  The index of the maximum probability(0.7) is 1.
  ```python
  acc = accuracy_fn(y, y_eval.argmax(dim=1)) # .argmax selects index of max probabality
  ```

if you do print statements like this you can see what goes on background
```python
print(f'Loss calculation: {loss_fn(y_eval, y)}')
loss += loss_fn(y_eval, y)
print('Actual label:', y)
print('Predicted label:', y_eval.argmax(dim=1))
acc = accuracy_fn(y, y_eval.argmax(dim=1))
```
